### DISCUSSION TITLE
Handle Conversation parameters

### DISCUSSION THREAD

**Discussion by DanielSpravtsev** [nutgram/nutgram#233] [Q&A]
Hey there!
In reference to:
https://github.com/nutgram/nutgram/blob/e21a8923373e84bda6ef1bdbfde182a259dbfba1/src/Conversations/Conversation.php#L111

When you using onCommand handler with Conversation class, you could also pass variables and handle it at first step method (by default it's `start`)
```
$bot->onCommand('add {name} {address}', AddConversation::class);
```

Conversation 

```
class AddConversation extends Conversation
{
    public function start(Nutgram $bot, ...$parameters)
    {
        $bot->sendMessage($parameters['name'] .' | '.$parameters['address']);
        $this->end();
    }
}
```

Hope someone wondered about this like me and now able to found a fast solution 😄

**Comment by Lukasss93** [comment]
Fast solution: just override the `$step` parameter with another method name like this:
```php
class AddConversation extends Conversation
{
    protected ?string $step = 'first';

    public function first(Nutgram $bot, ...$parameters)
    {
        $bot->sendMessage($parameters['name'] .' | '.$parameters['address']);
        $this->end();
    }
}
```

Result:
![immagine](https://user-images.githubusercontent.com/4071613/199022684-ca308607-844a-49c5-9178-089bb0bfcf77.png)

**Comment by sergix44** [comment]
```php
class AddConversation extends Conversation
{
    protected ?string $step = 'first';

    public function first(Nutgram $bot, string $name, string $address)
    {
        $bot->sendMessage("$name | $address");
        $this->end();
    }
}
```

\+ strict

**Reply by juankstylez** [reply]
ok, that's exactly what I'm doing, everything is correct in the first step,

but in the second step I need to return the user and ask the first step if a condition is not met.

I return it like this

public function secondStep(Nutgram $bot)
    {

// if is not a callback query, ask again!
         if (!$bot->message()->text) {
          $this->firsStep($bot);
           return;
        }
    }    
       When I return it like this (in the second step) it gives me the error that the first function, I mean the first step, was expecting two arguments.

The question:

How do I pass the argument I already received in the first step to the second step? so that ? for when the Second Step is met the error condition returns me to the first step with the argument that the function receives (in the first step)

**Reply by juankstylez** [reply]
Solved With CHATGPT ! <3 

```
class AddConversation extends Conversation
{
    protected ?string $step = 'first';
    protected ?string $name = null;

    public function primero(Nutgram $bot, string $name)
    {
        $this->name = $name; // Store the name in the instance variable
        $bot->sendMessage("$name");
        $this->next('segundo');
    }

    public function segundo(Nutgram $bot)
    {
        $bot->sendMessage("Hello");
        // If it's not a callback query, ask again
        if (!$bot->message()->text) {
            $this->primero($bot, $this->name); // Use the instance variable to pass the name
            return;
        }
    }
}
```
In the above example, the received name was stored in the instance variable $name. Then, in the second function segundo, $name is accessed via $this->name. When calling the primero function again, $this->name is passed as the second argument, allowing the function to access the originally received name.
